Tests/LibDNS/TestDNSResolver.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2025, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/Socket.h>
+#include <LibDNS/Resolver.h>
+#include <LibFileSystem/FileSystem.h>
+#include <LibTLS/TLSv12.h>
+#include <LibTest/TestCase.h>
+
+TEST_CASE(test_udp)
+{
+    Core::EventLoop loop;
+
+    DNS::Resolver resolver {
+        [&] -> ErrorOr<DNS::Resolver::SocketResult> {
+            Core::SocketAddress addr = { IPv4Address::from_string("1.1.1.1"sv).value(), static_cast<u16>(53) };
```

Comment:
I don't super love that these tests reach out to the internet, but sure I guess; it's fine as an integration test.